core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -471,13 +477,26 @@ private static MetadataUpdate readRemoveSnapshotRef(JsonNode node) {
   }
 
   private static MetadataUpdate readSetProperties(JsonNode node) {
-    Map<String, String> updated = JsonUtil.getStringMap(UPDATED, node);
-    return new MetadataUpdate.SetProperties(updated);
+    Map<String, String> updates;
+    // first read UPDATED for backward compatibility
+    if (node.has(UPDATED)) {
```

Comment:
nit: I think you should always check the correct property first, so we don't prioritize the obsolete one.  This is pretty minor because you likely wouldn't have both.

Reply:
I've slightly adjusted the code so that in the unlikely event that both fields are defined, we take the new one